services/cannon/src/Cannon/API.hs

```diff
@@ -111,6 +112,9 @@ sitemap = do
     post "/i/bulkpush" (continue bulkpush)
         request
 
+    get "/i/presences/:user/:conn" (continue checkPresence) $
```

Comment:
Actually, since it's not like the `Presence` is returned, what about `/i/checkpresence/`?

Reply:
How is this end-point access-controlled?  (Out of curiosity; there is probably docs on this somewhere?)

Reply:
> Should it be /i/presence/ then?
> Actually, since it's not like the Presence is returned, what about /i/checkpresence/?

I'd rather keep `/i/presences/...` to keep things consistent with the rest of the API, e.g., `/users/:user` or `/conversations/:conv`.
However, you are totally right that it returns nothing, just 200/404 so a `HEAD` makes more sense which I just fixed https://github.com/wireapp/wire-server/pull/255/commits/073fc7844b598d280b78ad926d8618ca2651ba28 does that make sense?

> How is this end-point access-controlled? (Out of curiosity; there is probably docs on this somewhere?)

By convention, all endpoints that start with `/i/` do not get routed from the outside world, i.e., nginx does not upstream any such requests so you need to be inside the DMZ to access them; that being said, when inside the host, you can query any endpoint.